lib/utils/googleAuth.ts

```diff
@@ -133,12 +133,16 @@ export const hasReadOnlyPermissionOnly = (
 };
 
 export const getRedirect = (
-  user?: HackneyGoogleUserWithPermissions
+  user?: HackneyGoogleUserWithPermissions,
+  checkForReadOnlyPermissions?: boolean
 ): string | undefined => {
   if (!user) {
     return '/login';
   }
-  if (!hasAnyPermissions(user)) {
+  if (
+    !hasAnyPermissions(user) ||
+    (checkForReadOnlyPermissions && hasReadOnlyPermissionOnly(user))
+  ) {
```

Comment:
Perhaps, this would be better? Then all the hardcoded `true` arguments and the param can be avoided.
```ts
if (!hasAnyPermissions(user) || hasReadOnlyPermissionOnly(user)) { /*...*/ }
```

Reply:
nvm, see the line 136 comment

Reply:
Because the `getRedirect` function is shared between different page types at the moment it's sometimes fine for the user to have read only permissions. For example they can view the application by accessing the view application page, but they can't edit the application by accessing the edit application page. Both pages use the same function and flow to check whether redirect is required. With this setup the responsibility for the check is in the calling page, hence the new argument. We are looking at refactoring the whole authorization flow to make things easier to maintain and also to shift the responsibility of the logic into a single module.